src/plugins/plotlyPlot/components/PlotlyViewLayout.vue

```diff
@@ -0,0 +1,221 @@
+<template>
+<div class="l-view-section"></div>
+</template>
+
+<script>
+import Plotly from 'plotly.js-dist';
+import moment from 'moment'
+
+export default {
+    inject: ['openmct', 'domainObject'],
+    data: function () {
+
+        return {
+            telemetryObjects: [],
+            bounds: this.openmct.time.bounds(),
+            timeRange:  0,
+            plotData: {},
+            subscriptions: {}
+        }
+    },
+    mounted() {
+        this.plotElement = document.querySelector('.l-view-section');
+
+        this.composition = this.openmct.composition.get(this.domainObject);
+        this.composition.on('add', this.addTelemetry);
+        this.composition.on('remove', this.removeTelemetry);
+        this.composition.load();
+
+        this.openmct.time.on('bounds', this.refreshData);
+        this.openmct.time.on('clock', this.changeClock);
+    },
+    destroyed() {
+        this.unsubscribe();
+    },
+    methods: {
+        changeClock() {
+            if (this.openmct.time.clock()) {
+                Plotly.purge(this.plotElement);
+                this.telemetryObjects.forEach((telemetryObject, index) => {
+                    this.subscribeTo(telemetryObject, index);
+                });
+            }
+        },
+        addTelemetry(telemetryObject) {
+            this.telemetryObjects.push(telemetryObject);
+            const index = this.telemetryObjects.findIndex(obj => obj === telemetryObject);
```

Comment:
Since you are pushing to `this.telemetryObjects` you can find index by `this.telemetryObjects.length`